nautobot/extras/plugins/__init__.py

```diff
@@ -358,6 +375,47 @@ def register_jobs(class_list):
     # That is done in response to the `nautobot_database_ready` signal, see nautobot.extras.signals.refresh_job_models
 
 
+class PluginFilterExtension:
+    """Class that may be returned by a registered Filter Extension function."""
+
+    model = None
+
+    filterset_fields = {}
+
+    filterform_fields = {}
+
+
+def register_filter_extensions(filter_extensions, plugin_name):
+    """
+    Register a list of PluginFilterExtension classes
+    """
+    from nautobot.utilities.utils import get_filterset_for_model
+
+    for filter_extension in filter_extensions:
+        if not issubclass(filter_extension, PluginFilterExtension):
+            raise TypeError(f"{filter_extension} is not a subclass of extras.plugins.PluginFilterExtension!")
+        if filter_extension.model is None:
+            raise TypeError(f"PluginFilterExtension class {filter_extension} does not define a valid model!")
+
+        for new_filter_name, new_filter_field in filter_extension.filterset_fields.items():
+            if not new_filter_name.startswith(f"{plugin_name}_"):
+                raise ValueError(
+                    f"Attempted to create a custom filter `{new_filter_name}` that did not start with `{plugin_name}`"
+                )
+
+            get_filterset_for_model(filter_extension.model).add_filter(new_filter_name, new_filter_field)
+
+        for new_filter_name, new_filter_field in filter_extension.filterset_fields.items():
+            if not new_filter_name.startswith(f"{plugin_name}_"):
+                raise ValueError(
+                    f"Attempted to create a custom filter `{new_filter_name}` that did not start with `{plugin_name}`"
+                )
+
+            get_filterset_for_model(filter_extension.model).add_filter(new_filter_name, new_filter_field)
```

Comment:
Am I seeing double here, or is there some difference between these two `for` loops that I'm missing?

Reply:
Weird copy 🍝 issue I think when I ran black before saving changes.